src/main/scala/org/scalameter/Parameters.scala

```diff
@@ -1,20 +1,31 @@
 package org.scalameter
 
+import scala.language.existentials
 import scala.collection.{Iterable, immutable}
+import Parameters.P
+
 
 @SerialVersionUID(4203959258570851398L)
-case class Parameters(axisData: immutable.ListMap[String, Any]) {
+case class Parameters(axisData: immutable.ListMap[Parameter[P], P]) {
   def ++(that: Parameters) = Parameters(this.axisData ++ that.axisData)
-  def apply[T](key: String) = axisData.apply(key).asInstanceOf[T]
-  def map(f: ((String, Any)) => (String, Any)) = Parameters(axisData.map(f))
+  def apply[T: Manifest](key: String) = axisData.apply(Parameter[T](key)).asInstanceOf[T]
+  def map(f: (String, P) => (String, P)) = {
+    val newAxisData = axisData.map { case (k, v) =>
+      val n = f(k.name, v)
+      (Parameter(n._1)(k.typeHint), n._2)
+    }
+    copy(axisData = newAxisData.asInstanceOf[immutable.ListMap[Parameter[_], Any]])
+  }
 
   override def toString = s"Parameters(${axisData.map(t => t._1 + " -> " + t._2).mkString(", ")})"
 }
 
 object Parameters {
-  def apply(xs: (String, Any)*) = new Parameters(immutable.ListMap(xs: _*))
+  private type P = T forSome { type T }
```

Comment:
Please, avoid using existential types. We want our code robust against future possible changes, and the benefit from having them here is not high.